Pcp LMS Updates

peteS-UK opened the pull request:
I've made a number of updates to make this more specific for LMS environments.  Changes are as follows

Read the LMS IP address from PCP.CFG as default.  If the LMS server is non-discoverable, the address will be specified here by default.

If no LMS IP in PCP, attempt to discover the LMS IP through broadcasting on UDP - same way as squeezelite discovers.  If both these methods fail, then use the lms/config.json or default to local.  The local default will likely not be used since discovery will find the local server anyway.

I found the current behaviour of 3 layers of screensaver very confusing.  I added detection of the power state of the squeezelite player and used the clock as the screen display when the player is off, then used the snake and blank screen savers as before.

The reading of the values from the config failed - it was taking item.value as if it was reading from a web page, not just item.

Changed the spelling of iddle to idle (since the rest of the var names are in english)

Read and use existing values of login.json if it exists

Add a script to create config.js through readline questions

Add date and time formats to config.json

I also created a small .js to automatically change contrast on sunrise/sunset and use the http server to update the contrast - happy to share if you like the idea.


If the contrast is changed whilst the screen is off, it's not updated.  So, reset the contrast on screen wake

Format the values for samplerate for DSD values

Add samplesize metdata and format correctly

audiophonics commented:
**Wow !** 

I never thought someone would upgrade this so fast. 
You perfectly got the idea of what makes this code imperfect : it was initialy written for a RaspDacMini on Volumio, then was ported to the EVO and then again ported to pCp.

Not sure I can test all of this before end of next week but at a glance everything looks fine. 

So far the only thing mildly bothering me is the ` formatSamplerate()` function that I would have defined as a method of the `_Player` object in lms.js rather than a function in index.js (I try to keep index.js as much distribution-agnostic as possible).
But if everything else is alright I will move that bit myself in a next update.

I agree that the **clock -> screensaver -> deepsleep** part is way too complex and unsettling. 
Also all feedback I ever got about the screensaver was people disliking the snake animation anyway, so simplifying that part is already on my todo-list.


Why not look at the sunset thing / http config page ! 
I was thinking of dropping some custom files in /www/cgi-bin to expose some user-friendly options for configuration as well but maybe you have a better way of achieving this. 


A big thank you.

peteS-UK commented:
Hi

I had a bit of a headstart because I’d taken the old python2 version from a while back and changed it to python3 and rewritten most of it a while back (peteS-UK/EvoSabre-DAC-PCP: Script and extensions required to configure piCorePlayer 8 on a Audiophonics EVO-SABRE 2xES9038Q2M DAC (github.com)<https://github.com/peteS-UK/EvoSabre-DAC-PCP>) so I kinda knew what I was trying to do.  I was just taking a look at my python stuff to see what I’d need to do to rebuild it for pcp9, which is in RC3 now.  Before doing that, I thought I’d just take a look and see if you’d published anything new, so just good timing 😊

I’m running it on pcp 8 64 bit, so all works fine on 64, and I’m sure it’ll be fine on pcp 9 as well – I’ll try it as soon as pcp 9 is released.

Yes, I just put formatSamplerate and formatSamplesize where I used them.  I’ve moved them to methods and pushed new index.js and lms.js to my fork.

I didn’t add the sunset stuff to the existing files as I wasn’t sure if you’d want to include it.  I haven’t externalised the contrast and location, but attached is the sun.js file I’m using – if you want to integrate it, it’s pretty simple I guess.

Yes, it’d be good to put up a simple webpage for all of the configs – I took a look at the server you had on one of the other repos but haven’t had a chance to do anything with it (and I’m not much cop at HTML)

With the IR stuff, you can use a direct interface on squeezelite rather than pcp.  I’ll post the files.  I wondered why you were using the shutdown script rather than using the GPIOs and the built in shutdown stuff in the pcp tweaks.  Seems to work fine doing that.

BTW, when the readme is updated, it would be good to include a couple of the squeezelite settings as from my readme – e.g. DOP, digital volume control etc..

Let me know if there’s anything I can do – I’m doing this for myself anyway 😊

Pete S

audiophonics commented:
I just tested the whole thing, everything is great. 

I need to look more into alternatives to basic shutdown script, sadly you get used to systemd easily managing everything for you. Will try to add that & more documentation by the end of April. 

Thank you again.

peteS-UK commented:
Hi

Good to hear that it's all working - hopefully it'll be useful for others.  My understanding originally is that on shutdown from the remote, GPIO 17 is active low and pulled up.  You can just set that in the UI.  I haven't tried to do it through modifying pcp.cfg, although that's no doubt possible.

Anyhow, glad it's working and let me know when you get to look at it again.  I'll let you know if I do anything else to it in the meantime.

Peter


From: Audiophonics ***@***.***>
Sent: Friday, March 8, 2024 2:49 PM
To: audiophonics/Evo_Sabre ***@***.***>
Cc: peteS-UK ***@***.***>; Author ***@***.***>
Subject: Re: [audiophonics/Evo_Sabre] Pcp LMS Updates (PR #4)


I just tested the whole thing, everything is great.

I need to look more into alternatives to basic shutdown script, sadly you get used to systemd easily managing everything for you. Will try to add that & more documentation by the end of April.

Thank you again.

-
Reply to this email directly, view it on GitHub<https://github.com/audiophonics/Evo_Sabre/pull/4#issuecomment-1985826521>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/API7QEOSH7IPZOUL2FX2SFLYXHFUFAVCNFSM6AAAAABD6DYJZSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBVHAZDMNJSGE>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>
